subsys/bootloader/include/fw_metadata.h

```diff
@@ -86,6 +121,29 @@ static inline bool memeq_32(const void *expected, const void *actual,
 	return true;
 }
 
+static inline bool memeq_8(const void *expected, const void *actual, u32_t len)
+{
+	const u8_t *expected_8 = (const u8_t *) expected;
+	const u8_t *actual_8   = (const u8_t *) actual;
+
+	for (u32_t i = 0; i < len; i++) {
+		if (expected_8[i] != actual_8[i]) {
+			return false;
+		}
+	}
+	return true;
+}
+
+static inline bool memeq(const void *expected, const void *actual, u32_t len)
+{
+	if (((u32_t)expected & 3) || ((u32_t)actual & 3) || ((u32_t)len & 3)) {
```

Comment:
This looks a lot like a stdlib memeq, could you verify that this is not an exact duplicate of how the standard `<string.h>` implementation does it?

Reply:
On second thought, we do not have full control over if that code will change, so IMO it makes sense to roll or own version regardless of if its a duplicate.